src/transport_performance/gtfs/routes.py

```diff
@@ -131,3 +137,50 @@ def scrape_route_type_lookup(
     route_lookup = pd.DataFrame(zip(cds, txts), columns=["route_type", "desc"])
 
     return route_lookup
+
+
+def get_saved_route_type_lookup(
+    path: Union[str, pathlib.Path] = pathlib.Path(
+        "tests/data/gtfs/route_lookup.pkl"
+    )
+) -> pd.DataFrame:
+    """Get the lcoally saved route type lookup as a dataframe.
+
+    Parameters
+    ----------
+    path : Union[str, pathlib.Path], optional
+        The path to the route type lookup,
+        by default pathlib.Path("tests/data/gtfs/route_lookup.pkl")
+
+    Returns
+    -------
+    pd.DataFrame
+        The route type lookup
+
+    """
+    # defences
+    _is_expected_filetype(
+        pth=path, param_nm="path", check_existing=True, exp_ext=".pkl"
+    )
+    lookup = pd.read_pickle(path)
+    ACCEPTED_TYPES = (dict, pd.DataFrame)
+    # check unserialized .pkl file is of correct type
+    if not isinstance(lookup, ACCEPTED_TYPES):
+        raise TypeError(
+            "Serialized object in specified .pkl file is of type: "
+            f"{type(lookup)}. Expected {ACCEPTED_TYPES}"
+        )
+    # convert dicts to pandas df
+    if isinstance(lookup, dict):
+        lookup = pd.DataFrame(lookup)
+    if len(lookup) < 1:
+        warnings.warn("Route type lookup has length of 0", UserWarning)
+    EXPECTED_COLS = ["route_type", "desc"]
+    # check columns
+    for col in lookup.columns.values:
+        if col.lower() not in EXPECTED_COLS:
+            warnings.warn(
+                f"Unexpected column '{col}' in route type lookup", UserWarning
+            )
```

Comment:
This may be refactored to:
```
    for col in lookup.columns.values:
        _check_column_in_df(lookup, col)
```

Reply:
As far as I'm aware, that function raises errors. In this case we are only raising a warning.